src/routes/LPOptimisers/useLPVaults.ts

```diff
@@ -0,0 +1,49 @@
+import { providers } from 'ethers';
+import { useEffect } from 'react';
+
+import {
+  initialiseVaultsForSignerThunk,
+  initialiseVaultsThunk,
+} from '../../store/features/ecosystem';
+import { useAppDispatch, useAppSelector } from '../../store/hooks';
+
+export const useLPVaults = (signer: providers.JsonRpcSigner | null) => {
+  const vaultsLoaded = useAppSelector((state) => state.ecosystem.vaultsLoadedState);
+  const signerLoadedState = useAppSelector((state) => state.ecosystem.signerLoadedState);
+  const shouldInitVaults = vaultsLoaded === 'idle';
+  const vaultsInitialised = vaultsLoaded === 'succeeded';
+  const vaultsInitialisedWithSigner = signerLoadedState === 'succeeded';
+  const lpVaults = useAppSelector((state) => state.ecosystem.lpVaults);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    if (!shouldInitVaults) {
+      return;
+    }
+    void dispatch(initialiseVaultsThunk());
+  }, [shouldInitVaults]);
+
+  useEffect(() => {
+    // todo: Costin can we unload vaults if user logs out?
```

Comment:
Yes, you can build the vault objects again and run only initialiseVaultsThunk.

Reply:
fixed in 6ef5bcf06c04cca74ef27b1c76fe1b2690538f89